paddle/fluid/inference/tests/api/CMakeLists.txt

```diff
@@ -105,9 +105,10 @@ function(inference_analysis_api_object_dection_int8_test_run TARGET_NAME test_bi
 	     --iterations=1)
 endfunction()
 
+# (aosewski) link to 'flags' to get access to 'onednn_use_input_mem_format' global flag.
 function(inference_analysis_api_test_with_fake_data_build TARGET_NAME filename)
 	inference_analysis_test_build(${TARGET_NAME} SRCS ${filename}
-        EXTRA_DEPS ${INFERENCE_EXTRA_DEPS})
+        EXTRA_DEPS ${INFERENCE_EXTRA_DEPS} flags)
 endfunction()
```

Comment:
> Could you advise on how to link analyzer_image_classification_tester.cc with flags.cc file where I have defined onednn_use_input_mem_format flag? I've tried to add flags target to the list of extra deps in CMakefile. Although this worked for me locally and in majority of CI, some CI checks fail with following errors.

I think you should not add flags target to the the list of extra deps in CMakefile. If you want to pass CI, how about ? 
```
void profile(bool use_mkldnn = false, bool onednn_use_input_mem_format = false) {
  if (use_mkldnn) {
      cfg.EnableMKLDNN();
      if (onednn_use_input_mem_format) 
        cfg.EnableOneDNNUseInputMemFormat();
    ....
  }
}

#ifdef PADDLE_WITH_MKLDNN
TEST(Analyzer_resnet50, profile_mkldnn) { profile(true /* use_mkldnn */, false /* onednn_use_input_mem_format */); }
TEST(Analyzer_resnet50, profile_mkldnn) { profile(true /* use_mkldnn */, true /* onednn_use_input_mem_format */); }
#endif
```